runtime/src/iree/base/loop_emscripten.js

```diff
@@ -14,28 +14,102 @@
 const LibraryLoopEmscripten = {
   $loop_emscripten_support__postset: 'loop_emscripten_support();',
   $loop_emscripten_support: function() {
-    class LoopEmscripten {
-      constructor() {
-        // TODO(scotttodd): store state here
-      }
+    const IREE_STATUS_OK = 0;
+    const IREE_STATUS_CODE_MASK = 0x1F;
+    const IREE_STATUS_ABORTED = 10 & IREE_STATUS_CODE_MASK;
+    const IREE_STATUS_OUT_OF_RANGE = 11 & IREE_STATUS_CODE_MASK;
+
+    class LoopOperation {
+      abort() {}
+    }
+
+    // IREE_LOOP_COMMAND_CALL
+    class LoopOperationCall extends LoopOperation {
+      constructor(scope, operationId, callback, user_data, loop) {
+        super();
 
-      loop_command_call(callback, user_data, loop) {
-        const IREE_STATUS_OK = 0;
+        this.callback = callback;
+        this.user_data = user_data;
+        this.loop = loop;
 
-        setTimeout(() => {
-          const ret =
-              Module['dynCall_iiii'](callback, user_data, loop, IREE_STATUS_OK);
+        this.timeoutId = setTimeout(() => {
+          getWasmTableEntry(callback)(user_data, loop, IREE_STATUS_OK);
```

Comment:
Still not sure if this is a stable way to call function pointers. It seems to work for some small cases but I see a few errors:

```js
// These appear to be equivalent, but both sometimes return
// `RuntimeError: table index is out of bounds`
getWasmTableEntry(callback)(user_data, loop, IREE_STATUS_OK);
{{{ makeDynCall('iiii', 'callback') }}} (user_data, loop, IREE_STATUS_OK);

// This is what I had before, but it sometimes returns
// `TypeError: Module.dynCall_iiii is not a function`
Module['dynCall_iiii'](callback, user_data, loop, IREE_STATUS_OK);
```

Should probably check with the Emscripten devs.

Reply:
Ehhhh, read through some PRs/Issues and settled on something that I think works... (use the generic `dynCall` with `-sDYNCALLS=1 -sEXPORTED_RUNTIME_METHODS=['dynCall']` linkopts)

Here's the latest source I'm referencing: https://github.com/emscripten-core/emscripten/blob/22b137742d40da3949a0e97cfc8e8172e41e2ace/test/test_core.py#L7098-L7120

Reply:
I think you need to declare it as a dep:
https://github.com/emscripten-core/emscripten/blob/1f0e614acf895d6b75d693227b99cd3c89a4f30e/ChangeLog.md?plain=1#L267-L269
so add to `DEFAULT_LIBRARY_FUNCS_TO_INCLUDE` or `EXPORTED_RUNTIME_METHODS` or better specify it on the method:
https://github.com/emscripten-core/emscripten/blob/1f0e614acf895d6b75d693227b99cd3c89a4f30e/src/library_addfunction.js#L165

Reply:
The CallFork test case is pretty unique in that it
1) In C++ (wasm) defines a callback function using a closure
2) In JS, that callback function is run within `setTimeout()`
3) The callback function itself defines a new callback function using a closure
4) In JS, the second callback function is run within `setTimeout()`

The other test cases don't have steps 3-4.

I'm thinking that somehow the `Module['dynCall'](` form is able to handle that reentrancy while `{{{ makeDynCall }}}` / `dynCall_iiii.apply` is not.

Source: https://github.com/iree-org/iree/blob/33ace455f664e93654cd60646d106a35a195a54c/runtime/src/iree/base/loop_test.h#L67-L95